src/__tests__/Button.types.test.tsx

```diff
@@ -0,0 +1,11 @@
+import React from 'react'
+import {Button} from '../Button'
+
+export function shouldAcceptOnlyAChildProp() {
```

Comment:
Could we also test one of the Primer props for accepting valid values and rejecting invalid values? eg `size`

Reply:
There's a bunch of tests - https://github.com/primer/react/blob/main/src/__tests__/Button.test.tsx that already validate that some props are accepted, but happy to add additional ones here

Reply:
45d5ec3244a5ac29f53fc8c2cc2a7778857f683d

Added here, which should validate all of the `ButtonProps` an aria/dom prop and the `onClick` handler's types I believe